sourdough/sourdough.py

```diff
@@ -380,6 +382,44 @@ def detectVSphereHost():
     except socket.error:
       this.logger.info('Cannot connect to vSphere hypervisor %s', hostname)
 
+def connectVcenter():
+  '''
+  Connect to Vcenter Server
+
+  Returns the UUID and Vcenter Connect objects
+  '''
+  uuid = getUUID()
+  loadSharedLogger()
+  secure=ssl.SSLContext(ssl.PROTOCOL_TLSv1)
+  secure.verify_mode=ssl.CERT_NONE
+
+  this.logger.debug('secure.verify_mode=ssl.CERT_NONE')
+
+  vSphereSettings = loadVSphereSettings()
+  if not vSphereSettings:
+    this.logger.error('Could not load vSphereSettings!')
+    return None
+  this.logger.debug('vSphereSettings: %r', vSphereSettings)
+
+  try:
+    hostname = vSphereSettings['hostname']
+    password = vSphereSettings['password']
+    username = vSphereSettings['username']
+  except KeyError:
+    this.logger.critical('Failed to load all parameters from cached vSphereSettings toml file')
+    return None
+
+  this.logger.debug('hostname=%s user=%s password=%s', hostname, username, password)
+
+  try:
+    si= SmartConnect(host=hostname, user=username, pwd=password, sslContext=secure)
+    this.logger.debug('SmartConnect succeeded')
+    searcher = si.content.searchIndex
+    this.logger.debug('Searching for VM for UUID %s', uuid)
+    vm = searcher.FindByUuid(uuid=uuid, vmSearch=True)
+    return uuid, si, vm
```

Comment:
Please return these in a dict so that the calling function doesn't have to be aware of the order these are returned.

Reply:
Please check 9c078c7a10f23335cec3283d95dd8d8b76f04921